tests/fillers/vm_fillers/paygas_fillers.py

```diff
@@ -327,11 +337,41 @@
             # "vyperLLLCode": [
             #     "seq",
             #     ["SSTORE", 0, 1],
-            #     ["PAYGAS", 1],
+            #     ["SSTORE", 2, ["ADD", 1, ["PAYGAS", 1]]],
             #     ["SSTORE", 1, 1],
             #     ["assert", 0],
             # ],
-            "code": b"`\x01`\x00U`\x01\xf5P`\x01`\x01U`\x00a\x00\x18W`\x00\x80\xfd[",
+            "code": b"`\x01`\x00U`\x01\xf5`\x01\x01`\x02U`\x01`\x01U`\x00a\x00\x1dW`\x00\x80\xfd[",
+        }
+    }),
+    expect(
+        networks=["Sharding"],
+        transaction={
+            "to": address,
+            "accessList": [
+                [address, b""],
+            ],
+        },
+        post_state=[
+            (address, "balance", 18951),
```

Comment:
Should the balance in post state be `pre_state_balance - transaction_gas_limit * gas_price`? As all gas will be confiscated.

Reply:
No, here state is reverted with the REVERT opcode which doesn't consume all gas. See https://github.com/ethereum/EIPs/blob/master/EIPS/eip-140.md

Reply:
Yep you are right. I mistake the `assert` here for the one in Solidity(`asset` there will take away all the gas).